cloudant-sync-datastore-core/src/main/java/com/cloudant/mazha/CouchClient.java

```diff
@@ -505,17 +523,10 @@ public Type getType() {
         Misc.checkNotNull(type, "Type");
 
         URI doc = uriHelper.documentUri(id, options);
-        InputStream is = null;
-        try {
-            HttpConnection connection = Http.GET(doc);
-            is = executeToInputStreamWithRetry(connection);
-            T returndoc = jsonHelper.fromJson(new InputStreamReader(is, Charset.forName("UTF-8"))
-                    , type);
-            logger.fine("getDocument returning " + returndoc);
-            return returndoc;
-        } finally {
-            closeQuietly(is);
-        }
+        HttpConnection connection = Http.GET(doc);
+        T returndoc = executeToJsonObjectWithRetry(connection, type);
+        logger.fine("getDocument returning " + returndoc);
```

Comment:
This is outside the scope of the change but I wonder whether these log lines should be guarded to avoid expensive `toString()` operations being done on large documents?